judge/views/contests.py

```diff
@@ -234,8 +234,23 @@ def get_context_data(self, **kwargs):
             .annotate(has_public_editorial=Sum(Case(When(solution__is_public=True, then=1),
                                                     default=0, output_field=IntegerField()))) \
             .add_i18n_name(self.request.LANGUAGE_CODE)
-        context['contest_has_public_editorials'] = any(
-            problem.is_public and problem.has_public_editorial for problem in context['contest_problems']
+        context['metadata'] = {
+            'has_public_editorials': any(
+                problem.is_public and problem.has_public_editorial for problem in context['contest_problems']
+            ),
+        }
+        context['metadata'].update(
+            **self.object.contest_problems
+            .annotate(
+                partials_enabled=F('partial').bitand(F('problem__partial')),
+                pretests_enabled=F('is_pretested').bitand(F('contest__run_pretests_only')),
```

Comment:
Can someone confirm that this will work even after systesting? cc @Xyene @quantum5

Reply:
This unfortunately won't be correct after systests, but I don't see any way to correctly grab that information after the settings are unchecked. FWIW this is also how it was done in #700, so it might be fine (?)

Reply:
This is a defect in the design of the pretest system. What we really want is two different booleans, but that's out of scope here.